server/src/main/java/org/elasticsearch/cluster/coordination/ClusterFormationFailureHelper.java

```diff
@@ -95,6 +97,7 @@ public void onFailure(Exception e) {
                 protected void doRun() {
                     if (isActive()) {
                         logger.warn(clusterFormationStateSupplier.get().getDescription());
+                        logLastFailedJoinAttempt.run();
```

Comment:
Perhaps these should be the other way round, so that the stack trace doesn't push the more descriptive message off the screen.

Reply:
https://github.com/elastic/elasticsearch/pull/39756/commits/765f66f8f407bde94a5aa01059cd498939ce11a0